cmd/object-handlers.go

```diff
@@ -620,7 +623,7 @@ func (api objectAPIHandlers) CopyObjectHandler(w http.ResponseWriter, r *http.Re
 		writeErrorResponse(w, ErrServerNotInitialized, r.URL)
 		return
 	}
-	if !objectAPI.IsEncryptionSupported() && crypto.S3KMS.IsRequested(r.Header) {
```

Comment:
We must not remove the `!objectAPI.IsEncryptionSupported()` because gateway backend could support SSE-KMS (AWS).
However the logic was wrong. Correct is: `!objectAPI.IsEncryptionSupported() || crypto.S3KMS.IsRequested(r.Header)`

Reply:
We can bring in later when gateway indeed supports such a functionality @aead

Reply:
Otherwise `!objectAPI.IsEncryptionSupported() || crypto.S3KMS.IsRequested(r.Header)` we are rejecting all Put/Copy operations if backend doesn't support encrypted.

Reply:
@harshavardhana is correct - for passing SSE-KMS headers to backend in the event it supports encryption, we will need to be able to distinguish we are running in gateway mode at the objectAPI interface

Reply:
Okay, agree. Actually the gateway functionality is totally broken - if a client sends SSE-KMS the backend does not see these headers even if it would support it (AWS) because of minio doesn't forward it. At least the client sees an error msg now.